src/Component/Attribute/AsGrid.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Grid\Attribute;
+
+#[\Attribute(\Attribute::TARGET_CLASS)]
+final class AsGrid
+{
+    public function __construct(
+        private readonly string $name,
```

Comment:
I think we should pass it as public. readonly is enough to tackle this.

Reply:
@diimpp public readonly is perfect. It's the modern way. We don't need getters for this kind of things.
Moreover, Symfony taclke this like that too.

Reply:
👍 for Loic perspective. 

Also, I would investigate if we can even make name optional, as we can already work with FQCN for grids. Name is still required for Sylius as a framework, but on end app, FQCN should be enough

Reply:
Should I change it back so ?

Reply:
Yep and indeed name can be optional and be the FQCN by default as you did for the filter attribute

Reply:
Done, tell me what you think